src/Relations/EmbedsOneOrMany.php

```diff
@@ -42,6 +43,8 @@ abstract class EmbedsOneOrMany extends Relation
      */
     public function __construct(Builder $query, Model $parent, Model $related, string $localKey, string $foreignKey, string $relation)
     {
+        parent::__construct($query, $parent);
```

Comment:
I think parent constructor should be called last. The affectation of `$this->query` and `$this->parent` can be removed, as the duplicate `$this->addConstraints();` at the end.

Reply:
Do you mean that those can be removed?

Reply:
Yes, it we call the parent constructor, we don't need to repeat all it does.